reactive/kotlinx-coroutines-jdk9/src/Await.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.jdk9
+
+import java.util.concurrent.Flow.Publisher
+import org.reactivestreams.FlowAdapters
+import kotlinx.coroutines.reactive.*
+
+/**
+ * Awaits for the first value from the given publisher without blocking a thread and
+ * returns the resulting value or throws the corresponding exception if this publisher had produced error.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this function
+ * immediately resumes with [CancellationException].
+ *
+ * @throws NoSuchElementException if publisher does not emit any value
+ */
+public suspend fun <T> Publisher<T>.awaitFirst(): T = FlowAdapters.toPublisher(this).awaitFirst()
+
+/**
+ * Awaits for the first value from the given observable or the [default] value if none is emitted without blocking a
+ * thread and returns the resulting value or throws the corresponding exception if this observable had produced error.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this function
+ * immediately resumes with [CancellationException].
+ */
+public suspend fun <T> Publisher<T>.awaitFirstOrDefault(default: T): T =
```

Comment:
Should these, too, be inline?

Reply:
There is nothing to inline

Reply:
@fvasco Actually, there's the `ContinuationImpl` to inline. You can even see that for suspending functions, there's no "nothing_to_inline" warning.

Reply:
@LouisCAD
Are you considered the tail call optimization?

Reply:
I'm not, but I expect inlining a suspending function like that optimizes further as there's no "nothing_to_inline" warning. Maybe @elizarov can confirm?

Reply:
It does not have to be inline because there is a tail-call optimization.